core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -1409,6 +1410,32 @@ class LogManager(logDirs: Seq[File],
       None
     }
   }
+
+  def readBrokerEpochFromCleanShutdownFiles(): Long = {
+    // Verify whether all the log dirs have the same broker epoch in their clean shutdown files. If there are dir is not
+    // live, fail the broker epoch check.
+    if (liveLogDirs.size < logDirs.size) {
+      return -1L
+    }
+    var brokerEpoch = -1L
+    for (dir <- liveLogDirs) {
+        val cleanShutdownFileHandler = new CleanShutdownFileHandler(dir.getPath)
+        var textBrokerEpoch = -1L
+        try {
+          textBrokerEpoch = cleanShutdownFileHandler.read
+        } catch {
+          case e: Throwable =>
+            info(s"loading broker epoch from ${dir.toPath} with exception ${e.toString}")
+            return -1L
+        }
+        if (brokerEpoch != -1 && textBrokerEpoch != brokerEpoch) {
+          info(s"Found different broker epochs a=$brokerEpoch vs b=$textBrokerEpoch")
```

Comment:
Could we log the dir names instead of using a and b?

Reply:
We don't track the dir names already iterated, so only log the current dir name here.